src/size.c

```diff
@@ -201,6 +209,32 @@ SEXP vctrs_recycle(SEXP x, SEXP size_obj, SEXP x_arg) {
   return vec_recycle(x, size, &x_arg_);
 }
 
+// [[ include("vctrs.h") ]]
+SEXP vec_recycle_fallback(SEXP x, R_len_t size, struct vctrs_arg* x_arg) {
+  if (x == R_NilValue) {
+    return R_NilValue;
+  }
+
+  // Can't use `vec_size()` here, as `x` might not be a vector
+  R_len_t x_size = vec_raw_size(x);
+
+  if (x_size == size) {
+    return x;
+  }
+
+  if (x_size == 1) {
+    SEXP subscript = PROTECT(Rf_allocVector(INTSXP, size));
```

Comment:
Do we need a searchable TODO in places where we could replace repetitions with altrep ones?

Reply:
Generally this might be a good idea. I don't think it would matter for this specific case because it falls back to `[` so it would immediately be expanded